### DISCUSSION TITLE
what equipment i need?

### DISCUSSION THREAD

**Discussion by ktnik-prog** [Open-Agriculture/AgIsoStack-plus-plus#343] [General]
Hello. I started learning isobus using your lessons. Tell me what physical equipment I need to purchase? 
I have a computer on which I can install Linux. 
I have a Timble fmx 1000 monitor and it looks like it is unlocked for isobus. 
What else do I need to buy for tests and training?
![msg536288700-69687](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/73779375/963f2eda-0a4d-4cbc-bd28-66e6cf2a6ad0)

**Comment by ad3154** [comment]
Hello there, and welcome to the community!

If your monitor is ISOBUS compatible, then you'll just need a USB to CAN adapter of some kind that works with SocketCAN for Linux, such as a [PEAK](https://www.peak-system.com/PCAN-USB.199.0.html?&L=1) adapter, or a [systec](https://www.systec-electronic.com/en/products/interfaces-gateways-amp-controls/sysworxx-usb-can-module1), or something else like that.

You'll also need a working CAN bus attached to the monitor, which can be very simple for testing, but needs at least one CAN terminator. It's honestly easiest if you have some ISOBUS machine you can plug into, but for the minimum needed it's essentially just connecting all CAN-H together, all CAN-L together, and a tee into a terminator. Something like this:
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/10929341/be618ccd-f39c-4d44-942f-d00ffc4f7674)

**Reply by ktnik-prog** [reply]
thank for you for your help. I ordered this module 
[fischl](https://www.fischl.de/usbtin/)
In the meantime, I started installing the library
the command: sudo ip link set can0 up type can bitrate 250000
displays the message: Cannot find device "can0"
Perhaps I need to insert the adapter first?
I have a lot of questions, is there any instructions for dummies? or installation video?

**Reply by ad3154** [reply]
> Perhaps I need to insert the adapter first?

Yes, that is true, but you will also need different setup for an SLCAN probe such as that one. Adapters like that one which do SLCAN may be less reliable than a more expensive one, but I hope that one works OK for you.

You will probably need to do this:
```
sudo modprobe slcan
sudo slcand -o -s5 -f /dev/ttyAMC0
sudo ip link set up can0
```

It is OK if you have questions!

**Comment by trm01** [comment]
Hi,

I've just started  playing around with some of the task controller client example code and specifically the changing around the  DDOP parameters. I find sitting in the cab of the tractor as a development environment quite inefficient.  I'm curious if anyone can recommend a virtual ISOBus Task Controller with (or without) GPS simulation that I can run on my desktop and connect to it from my AgIsoStack-plus-plus app? 

Thanks,
Tim

**Reply by GwnDaan** [reply]
Hey @trm01,

I recognize your struggles, I'm somewhat in the same boat as well. Though I have a detachable [Müller](https://mueller-elektronik.de/en/produkte/touch800-3/?category=482#produkt) monitor that I can take inside and hook up to a bench setup I cobbled together. 

If you've got a bit of money to spend, I heard people talking about the [CANoe from vector](https://www.vector.com/int/en/products/products-a-z/software/canoe/option-iso11783/#c33214) as a good virtual environment to use. Though that's a little out of my budget range for now hahah. Maybe in the future we can have our own virtual TC/VT server 🤔

**Reply by trm01** [reply]
Thanks for the suggestions. 
I received some quotes for the CANoe a while back and they we're quite high. I could buy probably buy a couple monitors for cheaper, so that is probably the direction I may go. 
I've started dabbling in some of the AEF conformance tools as I saw mention of simulators. My fingers are crossed that something in there will provide useful.
